azure-quantum/azure/quantum/optimization/problem.py

```diff
@@ -275,5 +279,30 @@ def is_large(self) -> bool:
         for term in self.terms:
             set_vars.update(term.ids)
 
-        return (len(set_vars) >= Problem.NUM_VARIABLES_LARGE
-                and len(self.terms) >= Problem.NUM_TERMS_LARGE)
+        return (
+            len(set_vars) >= Problem.NUM_VARIABLES_LARGE
+            and len(self.terms) >= Problem.NUM_TERMS_LARGE
+        )
+
+    def download(self):
+        """Dowloads an uploaded problem as an instance of 'Problem'"""
+        if not self.uploaded_blob_uri:
+            raise Exception(
+                "Problem must be uploaded before it can be downloaded"
+            )
+        contents = download_blob(self.uploaded_blob_uri)
+        return Problem.deserialize(contents, self.name)
+
+    def get_terms(self, id:int) -> List[Term]:
+        """ Given an index the function will return
+        a list of terms with that index
+        """
+        terms = []
+        if self.terms != []:
+            for term in self.terms:
+                if id in term.ids:
+                    terms.append(term)
+            return terms
+        else:
+            raise Exception("Please download the problem on the client" \
```

Comment:
nit: Adding "There are currently no terms in this problem." at the beginning might give more context. So we have something like

"There are currently no terms in this problem. Please download the problem on the client or add terms to the problem to perform this operation."